### PR TITLE
Add support for httpRequest Promise

### PR DESCRIPTION
**Pull request by yuchonghua**
添加支持[httpRequest](https://github.com/ElemeFE/element/pull/3387) Promise

Please makes sure these boxes are checked before submitting your PR, thank you!
* [x] Add support for httpRequest Promise
* [x] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

**Comment by baiyaaaaa** [comment]
xhr 本来就有异步回调的机制，不需要返回promise，而且，request 在这里预期是需要返回xhr的，在后面会增加 abort 终止上传的功能

**Comment by yuchonghua** [comment]
其实很多时候返回给到的不是一个xhr，但是返回的方法其实可以同时支持then和catch和abort

**Comment by yuchonghua** [comment]
@baiyaaaaa 
`
var _promise = ajax()
_promise. abort = function(){取消逻辑}
return _promise
`
其实现在很多请求都是同时支持then和catch和abort

**Comment by yuchonghua** [comment]
@Leopoldthecoder  @QingWei-Li 为什么pr感觉现在都没人鸟了呢？

File: packages/upload/src/upload.vue [hunk 1]
```diff
@@ -111,7 +113,12 @@ export default {
         onError: err => {
           this.onError(err, rawFile);
         }
-      });
+      };
+      const requestPromise = this.httpRequest(options);
+      /* global Promise:true */
+      if (typeof Promise !== 'undefined' && requestPromise instanceof Promise) {
```
Comment (baiyaaaaa) [inline packages/upload/src/upload.vue]: 只需要判断`if(requestPromise && requestPromise.then)`就行了，不是所有promise都是原声Promise的实例

**Comment by yuchonghua** [comment]
@baiyaaaaa 已经修改

**Pull request merged**